## In this file load our data and overview

In [1]:
# import library
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [5]:
df=pd.read_csv('economic_indicators_dataset_2010_2023.csv')

In [6]:
df.head()

Date Country  Inflation Rate (%)  GDP Growth Rate (%)  \
0  2010-01-31  Brazil                1.23                 0.69   
1  2010-01-31  France                6.76                 2.59   
2  2010-01-31     USA                7.46                 4.84   
3  2010-02-28  Brazil                5.43                 0.31   
4  2010-02-28  Canada                0.69                -0.52   

   Unemployment Rate (%)  Interest Rate (%)  Stock Index Value  
0                  10.48               7.71           21748.85  
1                   4.27               7.39           10039.56  
2                   2.64               6.39           13129.10  
3                   8.26               6.09           23304.58  
4                  11.92              -0.51           16413.03

In [7]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 500 entries, 0 to 499
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Date                   500 non-null    object 
 1   Country                500 non-null    object 
 2   Inflation Rate (%)     500 non-null    float64
 3   GDP Growth Rate (%)    500 non-null    float64
 4   Unemployment Rate (%)  500 non-null    float64
 5   Interest Rate (%)      500 non-null    float64
 6   Stock Index Value      500 non-null    float64
dtypes: float64(5), object(2)
memory usage: 27.5+ KB


In [8]:
df.isnull().sum()

Date                     0
Country                  0
Inflation Rate (%)       0
GDP Growth Rate (%)      0
Unemployment Rate (%)    0
Interest Rate (%)        0
Stock Index Value        0
dtype: int64

In [10]:
df.drop_duplicates(inplace=True)

In [11]:
df.duplicated().sum()

0

In [12]:
df.describe()

Inflation Rate (%)  GDP Growth Rate (%)  Unemployment Rate (%)  \
count           500.00000           500.000000             500.000000   
mean              5.08514             2.414780               6.906720   
std               2.95633             4.074762               2.982406   
min               0.00000            -5.000000               2.000000   
25%               2.52500            -1.170000               4.257500   
50%               5.20500             2.580000               6.865000   
75%               7.71000             5.445000               9.502500   
max               9.99000             9.930000              11.970000   

       Interest Rate (%)  Stock Index Value  
count         500.000000         500.000000  
mean            4.698180       20926.226540  
std             3.152877       11066.259233  
min            -0.980000        1017.270000  
25%             2.027500       11587.862500  
50%             4.975000       20996.210000  
75%             7.352500       30824.847500  
max            10.000000       39982.210000

### Summary for Inflation Rate:

Mean: `5.08%` — Inflation is generally moderate.

Standard Deviation: `2.96` — There is some variability.

Min-Max: From `0% to 9.99% `— Some values are very low, while others are quite high.

Median (50%): `5.21%` — The mean and median are close, meaning the data is fairly balanced.

25%-75%: Most of the time, inflation has been between `2.5% and 7.7%.`

### Summary of GDP Growth Rate

Mean: `2.41%` — Growth is relatively slow.

Standard Deviation: `4.07` — There is a lot of variability, meaning growth can be very low or very high at times/countries.

Min: `-5%` — There has been economic contraction (negative growth) in some areas.

Max: `9.93%` — There has been very good growth in some places.

Median:` 2.58%` — Close to the mean.

25%-75%: Between `-1.17%` and `5.45%` — Most of the time, growth is moderate to good.

### Summary of Unemployment Rate

Mean: `6.91%` — Unemployment is somewhat high.

Standard Deviation: ``2.98` — There is some variability.

Min: `2% `— In some places, unemployment is very low.

Max: `11.97% `— In some areas, unemployment is quite high.

Median: `6.87%` — Close to the mean.

25%-75%: Between `4.26% and 9.50%` — Most of the time, unemployment is moderate to high.

### Summary of Interest rates

Mean: `4.70% `— Interest rates are moderate.

Standard Deviation: `3.15 `— There is some variability.

Min: `-0.98%` — In some places, there are negative interest rates (economic stimulus).

Max: `10%` — In some areas, rates are quite high.

Median: `4.98% `— Close to the mean.

25%-75%: Between ``2.03% and 7.35%` — Most of the time, interest rates are moderate

### Summary of stock market

Mean: `20,926` — The stock market is large scale.

Standard Deviation: `11,066` — There is a lot of variability.

Min: `1,017` — In some places, the market is small.

Max: `39,982 `— In some areas, the market is very large.

Median: `20,996` — Close to the mean.

25%-75%: Between `11,588 and 30,825` — Most of the time, the market is moderate to large.